tools/pipelines/templates/build-docker-service.yml

```diff
@@ -575,10 +549,26 @@ extends:
           buildToolsVersionToInstall: ${{ parameters.buildToolsVersionToInstall }}
 
     - ${{ if eq(parameters.shouldReleaseDockerImage, true) }}:
-      - template: /tools/pipelines/templates/include-publish-docker-service.yml@self
-        parameters:
-          containerRegistry: $(containerRegistryConnection)
-          containerTag: $(containerTag)
-          containerRegistryUrl: $(containerRegistryUrl)
-          containerName: ${{ parameters.containerName }}
-          tagName: $(tagNameOnReleaseImage)
+      - stage: publish_docker_official
+        dependsOn: build
+        displayName: Publish Official Docker Image
+        condition: and(succeeded(), eq(variables['release'], 'release'))
+        variables:
+        - template: /tools/pipelines/templates/include-vars-docker.yml@self
+          parameters:
+            containerName: ${{ parameters.containerName }}
+            shouldPushDockerImage: ${{ parameters.shouldPushDockerImage }}
+            testBuild: ${{ parameters.testBuild }}
+        jobs:
+        - template: /tools/pipelines/templates/include-publish-docker-service-steps.yml@self
+          parameters:
+            environment: container-registry-public
+            # $(containerRegistryConnection) and $(containerRegistryUrl) come from the container-registry-info variable
+            # group and need to be specified as runtime variables (variables from variable groups apparently are never
+            # available "statically" at parse/compile time, so can't be used with template-expression syntax ( '${{ }}' )).
+            containerRegistry: $(containerRegistryConnection)
+            containerRegistryUrl: $(containerRegistryUrl)
+            containerTag: $(containerTag)
+            containerName: public/${{ parameters.containerName }}
+            ${{ if eq(parameters.releaseKind, 'docker') }}:
+              tagName: ${{ parameters.tagName }}
```

Comment:
This is basically a slightly adjusted copy-paste of what used to be in `include-publish-docker-service.yml`.